drivers/platform/maxim/max32650/maxim_usb_uart.c

```diff
@@ -0,0 +1,509 @@
+		enum_register_descriptor(ENUM_DESC_QUAL,
+					 (uint8_t *)&device_qualifier_descriptor, 0);
+	}
+
+	enum_register_descriptor(ENUM_DESC_STRING, lang_id_desc, 0);
+	enum_register_descriptor(ENUM_DESC_STRING, mfg_id_desc, 1);
+	enum_register_descriptor(ENUM_DESC_STRING, prod_id_desc, 2);
+	enum_register_descriptor(ENUM_DESC_STRING, serial_id_desc, 3);
+	enum_register_descriptor(ENUM_DESC_STRING, cdcacm_func_desc, 4);
+
+	/* Handle configuration */
+	enum_register_callback(ENUM_SETCONFIG, setconfig_callback, NULL);
+
+	/* Handle feature set/clear */
+	enum_register_callback(ENUM_SETFEATURE, setfeatureCallback, NULL);
+	enum_register_callback(ENUM_CLRFEATURE, clrfeatureCallback, NULL);
+
+	/* Initialize the class driver */
+	if (acm_init(&config_descriptor.comm_interface_descriptor) != 0) {
+		ret = -EFAULT;
+		goto error_usb;
+	}
+
+	/* Register callbacks */
+	MXC_USB_EventEnable(MAXUSB_EVENT_NOVBUS, event_callback, NULL);
+	MXC_USB_EventEnable(MAXUSB_EVENT_VBUS, event_callback, NULL);
+	acm_register_callback(ACM_CB_READ_READY, usb_read_callback);
+
+	usb_read_complete = 0;
+
+	/* Start with USB in low power mode */
+	usbAppSleep();
+
+	struct no_os_irq_init_param nvic_ip = {
+		.platform_ops = &max_irq_ops,
+	};
+
+	ret = no_os_irq_ctrl_init(&max_usb_uart->nvic, &nvic_ip);
+	if (ret)
+		goto error_usb;
+
+	struct no_os_callback_desc uart_rx_cb = {
+		.callback = usb_callback,
+		.event = NO_OS_EVT_USB,
+		.peripheral = NO_OS_USB_IRQ,
+		.handle = MXC_USBHS,
+	};
+
+	ret = no_os_irq_register_callback(max_usb_uart->nvic,
+					  param->irq_id,
+					  &uart_rx_cb);
+	if (ret)
+		goto error_irq;
+
+	ret = no_os_irq_enable(max_usb_uart->nvic, param->irq_id);
+	if (ret)
+		goto error_irq;
+
+	*desc = descriptor;
+
+	return 0;
+error_irq:
+	no_os_irq_ctrl_remove(max_usb_uart->nvic);
+error_usb:
+	no_os_free(max_usb_uart);
+error_max_usb_uart:
+	no_os_free(descriptor);
+
+	return ret;
+}
+
+/**
+ * @brief Free the resources allocated by max_uart_init().
+ * @param desc - The UART descriptor.
+ * @return 0 in case of success, errno codes otherwise.
+ */
+static int32_t max_uart_remove(struct no_os_uart_desc *desc)
+{
+	struct max_usb_uart_desc *max_usb_uart;
+
+	if (!desc)
+		return -EINVAL;
+
+	max_usb_uart = desc->extra;
+
+	no_os_irq_ctrl_remove(max_usb_uart->nvic);
+	no_os_free(desc->extra);
+	no_os_free(desc);
+
+	return 0;
+}
+
+/**
+ * @brief Not implemented.
+ * @param desc - The UART descriptor.
+ * @return -ENOSYS
+ */
+static uint32_t max_uart_get_errors(struct no_os_uart_desc *desc)
+{
+	return -ENOSYS;
+}
+
+/**
+ * @brief Maxim platform specific UART platform ops structure
+ */
+const struct no_os_uart_platform_ops max_usb_uart_ops = {
+	.init = &max_uart_init,
+	.read = &max_uart_read,
+	.write = &max_uart_write,
+	.read_nonblocking = &max_uart_read_nonblocking,
+	.write_nonblocking = &max_uart_write_nonblocking,
+	.get_errors = &max_uart_get_errors,
+	.remove = &max_uart_remove
+};
+
+#if (TARGET_NUM == 32650)
```

Comment:
Since we have different drivers for each MCU I think we can just use the appropriate branch here.

Reply:
yes, I will change this